vowpalwabbit/cb_algs.cc

```diff
@@ -49,9 +49,14 @@ void predict_or_learn(cb& data, single_learner& base, example& ec)
 {
   CB::label ld = ec.l.cb;
   cb_to_cs& c = data.cbcs;
-  c.known_cost = get_observed_cost(ld);
-  if (c.known_cost != nullptr && (c.known_cost->action < 1 || c.known_cost->action > c.num_actions))
-    std::cerr << "invalid action: " << c.known_cost->action << std::endl;
+  auto optional_cost = get_observed_cost_cb(ld);
+  if (optional_cost.first) { c.known_cost = optional_cost.second; }
+  else
+  {
+    c.known_cost = CB::cb_class{};
+  }
+  if (optional_cost.first && (optional_cost.second.action < 1 || optional_cost.second.action > c.num_actions))
```

Comment:
maybe here instead of using `optional_cost.second.action` using `c.known_cost.action` would be easier to understand?